layer/trace_layer.cpp

```diff
@@ -98,10 +98,33 @@ static VkInstance get_instance_handle(const void* handle)
     return (entry != instance_handles.end()) ? entry->second : VK_NULL_HANDLE;
 }
 
+// The vk_layerGetPhysicalDeviceProcAddr of the next layer in the chain.
+// Retrieved during instance creation and used to forward this layer's
+// GetPhysicalDeviceProcAddr() after unwrapping its VkInstance parameter.
+static std::mutex                    gpdpa_lock;
+static PFN_GetPhysicalDeviceProcAddr next_gpdpa = nullptr;
```

Comment:
Is it possible for an app to create two instances with different layer settings and thus a different GPDPA per instance?